src/core/thread/network_data_local.cpp

```diff
@@ -69,6 +90,21 @@ Error Local::AddOnMeshPrefix(const OnMeshPrefixConfig &aConfig)
     // of an IEEE 802.15.4 interface MUST have a length of 64 bits.
     VerifyOrExit(!aConfig.mSlaac || aConfig.mPrefix.mLength == OT_IP6_PREFIX_BITSIZE, error = kErrorInvalidArgs);
 
+    SuccessOrExit(error = ValidatePrefix(aConfig.GetPrefix()));
+    SuccessOrExit(error = ValidatePreference(aConfig.mPreference));
+
+    // Check if the prefix overlaps mesh-local prefix.
+    VerifyOrExit(!aConfig.GetPrefix().ContainsPrefix(Get<Mle::MleRouter>().GetMeshLocalPrefix()),
+                 error = kErrorInvalidArgs);
+
```

Comment:
The logic here is same as `ValidateHasRoutePrefix` . Use a common method for this validation?